UnityProject/Assets/Scripts/Core/Chat/ChatUI.cs

```diff
@@ -323,10 +326,19 @@ private void UpdateMe()
 		public void AddChatEntry(string message, TMP_SpriteAsset languageSprite = null)
 		{
 			// Check for chat entry duplication
-			if (allEntries.Count > 0 && message.Equals(allEntries[allEntries.Count - 1].Message))
+			if (allEntries.Count > 5)
 			{
-				allEntries[allEntries.Count - 1].AddChatDuplication();
-				return;
+				for (int i = 0; i < 6; i++)
+				{
+					var entryToCheck = allEntries[allEntries.Count - i - 1];
+					string cleanedEntryMessage = Regex.Replace(entryToCheck.Message, @"<size=[^>]+>|</size>", string.Empty);
+					string cleanedMessage = Regex.Replace(message, @"<size=[^>]+>|</size>", string.Empty);
```

Comment:
is this done client side  since this might be slow?

Reply:
This is client sided.